fix(ivy): incorrectly remapping certain properties that refer to inputs

crisbeto opened the pull request:
During build time we remap particular property bindings, because their names don't match their attribute equivalents (e.g. the property for the `for` attribute is called `htmlFor`). This breaks down if the particular element has an input that has the same name, because the property gets mapped to something invalid.

The following changes address the issue by mapping the name during runtime, because that's when directives are resolved and we know all of the inputs that are associated with a particular element.

crisbeto committed:
fix(ivy): incorrectly remapping certain properties that refer inputs

During build time we remap particular property bindings, because their names don't match their attribute equivalents (e.g. the property for the `for` attribute is called `htmlFor`). This breaks down if the particular element has an input that has the same name, because the property gets mapped to something invalid.

The following changes address the issue by mapping the name during runtime, because that's when directives are resolved and we know all of the inputs that are associated with a particular element.

kara commented on packages/core/test/render3/properties_spec.ts:
```diff
@@ -77,6 +76,26 @@ describe('elementProperty', () => {
     expect(fixture.html).toEqual('<span id="_otherId_"></span>');
   });
 
+  it('should map properties whose names do not correspond to their attribute names', () => {
```
We're trying to avoid adding new tests with handwritten generated code. Can you convert these tests to TestBed tests and add them to the `core/test/acceptance` folder?  Also seems like we'll need a test in the compliance spec to test that we are generating code without the mapping.

kara reviewed (changes requested):
Changes look good to me, just needs some test updates

crisbeto commented:
@kara I've switched the tests over to `TestBed` and added a new test for the code generation. Can you take another look? The CI failures seem to be unrelated.

kara commented on packages/compiler-cli/test/compliance/r3_view_compiler_template_spec.ts:
```diff
@@ -602,4 +602,46 @@ describe('compiler compliance: template', () => {
     expectEmit(result.source, template, 'Incorrect template');
   });
+
+  it('should not remap property names whose names do not correspond to their attribute names',
```
Would you mind moving this to the compliance_binding_spec? This file is specifically for nested template tests, to the test doesn't really fit in here.

https://github.com/angular/angular/blob/6ed7120be0b284c302b786c567024d1ee754c747/packages/compiler-cli/test/compliance/r3_view_compiler_binding_spec.ts

crisbeto replied:
Done.

kara replied:
Thanks!

kara reviewed (approved):
~~LGTM~~

kara reviewed (changes requested):
@crisbeto Just realized that this PR should fix a number of Material tests, but we're not actually removing anything from the blocklist.  Can you remove the relevant tests?

https://github.com/angular/angular/blob/master/tools/material-ci/angular_material_test_blocklist.js#L1861

crisbeto commented:
Updated @kara. It also has two new bonus failures that are due to static queries.

filipesilva reviewed (approved):
LGTM for the changes in `tools/material-ci/angular_material_test_blocklist.js` (the ones I assumed triggered the `angular/fw-dev-infra` review).

kara reviewed (approved):
LGTM, though it needs a rebase

crisbeto commented:
It's rebased @kara.

kara commented:
[presubmit](http://test/OCL:234871613:BASE:234871688:1550702469282:d518b76b)

crisbeto committed:
fixup! fix(ivy): incorrectly remapping certain properties that refer to inputs
